lib/tlsca/ca.go

```diff
@@ -110,37 +128,82 @@ func (id *Identity) Subject() (pkix.Name, error) {
 	subject.Organization = append([]string{}, id.Groups...)
 	subject.OrganizationalUnit = append([]string{}, id.Usage...)
 	subject.Locality = append([]string{}, id.Principals...)
+
+	// DELETE IN (4.4.0)
+	// Groups are marshaled to both ASN1 extension
+	// and old Province section, for backwards-compatibility,
+	// however begin migration to ASN1 extensions in the future
+	// for this and other properties
 	subject.Province = append([]string{}, id.KubernetesGroups...)
 	subject.StreetAddress = []string{id.RouteToCluster}
 	subject.PostalCode = []string{string(rawTraits)}
 
+	for i := range id.KubernetesUsers {
+		kubeUser := id.KubernetesUsers[i]
+		subject.ExtraNames = append(subject.ExtraNames,
```

Comment:
If we're moving data out of the subject, why not move this information to the extensions? That where the SAN, key usage, and other metadata now lives.

Reply:
what's the benefit of doing that vs keeping it in subject?

Reply:
with keeping it in subject, there is a benefit of locality as we plan on keeping other things like common name to be used for groups, unless we want to completely stop using the subject.

Reply:
I thought that is what was being suggested in one of the comments, to get everything off of `subject.{Organization, OrganizationUnit, Locality}` in the future?

I don't have a technical argument for moving data out of the subject, but more for debugging. With our current approach of packing everything into the subject, when running a command like `openssl x509 -in <cert> -text -noout` I have to mentally map fields in the subject to Teleport metadata like `OU=groups`.

Ideally we'd see something like `x-teleport-groups: admin, dev` instead. It appears that's possible with extensions.

Reply:
I'd suggest we keep the metadata in the subject for now, but instead use name extensions instead of common fields. The same command you have outlined works with subject as well, no problem

Reply:
:+1: